server/src/main/scala/com/cloudera/livy/server/recovery/SessionRecovery.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Cloudera, Inc. under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Cloudera, Inc. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.cloudera.livy.server.recovery
+
+import com.cloudera.livy.{LivyConf, Logging}
+import com.cloudera.livy.server.batch.{BatchRecoveryMetadata, BatchSession}
+import com.cloudera.livy.sessions.{Session, SessionManager}
+import com.cloudera.livy.sessions.Session.RecoveryMetadata
+
+object SessionRecoveryMode {
+  val OFF = "off"
+  val RECOVERY = "recovery"
+}
+
+class SessionRecovery(sessionStore: SessionStore, livyConf: LivyConf) extends Logging {
```

Comment:
It kinda feels to me that it would be cleaner if this class didn't exist. e.g., instead of this, you instantiate a state store, and provide it in the constructor of the `SessionManager` instance, which also takes the session type as an argument.

That constructor then asks the state store for the existing sessions as part of its initialization.

Or it could be a method in `SessionStore`.

Basically I just don't see this adding much value as a separate class.